src/libtorch.cc

```diff
@@ -942,6 +942,16 @@ ModelInstanceState::ReadOutputTensors(
       batchn_shape.push_back(*itr);
     }
 
+    if (batchn_shape.size() == 0) {
+      RESPOND_ALL_AND_RETURN_IF_ERROR(
+          responses, request_count,
+          TRITONSERVER_ErrorNew(
+              TRITONSERVER_ERROR_INVALID_ARG,
+              (std::string("output '") + name +
+               "' is a scalar which is not supported.")
+                  .c_str()));
+    }
```

Comment:
Can we just prepare it as output tensor with shape [1]?

Reply:
While that would solve the issue, it would not be true to the actual output from the model.

Are we willing to sacrifice on the accuracy of output shape and allow scalar outputs by doing so? 
@deadeyegoodwin  @tanmayv25 I'd appreciate both of your opinion on the matter. 

On one hand we can avoid asking users to trace their model. If other backends allow scalar outputs do we extend this behaviour to them as well and document the same? 

If we do this on the output side to allow supporting scalars what stops us from doing the same for input scalars?

Reply:
I too believe if we can support the execution of scalar tensors by automatically reshaping the tensors then we should definitely do so. 
A user who doesn't have control over the model will surely appreciate it. Actually, they should not care about whether Triton is treating scalar inputs as [1] tensors as long as they can load the model and make an inference call to it. 
They can use reshape() in our model_config to support scalars on input size. 

However, the behavior should be consistent across all the backends. As we don't have a proper support for scalar tensors. I am fine returning this error for now.